src/cli/cmd/apply/mod.rs

```diff
@@ -0,0 +1,200 @@
+
+                run_script(
+                    script_path,
+                    Some(DARWIN_REBUILD_ACTION.to_string()),
+                    NIX_DARWIN_SCRIPT,
+                )
+                .await?;
+            }
+            #[cfg(target_os = "linux")]
+            System::NixOs(NixOs { action, .. }) => {
+                let profile_path =
+                    apply_path_to_profile(NIXOS_PROFILE, &resolved_path.store_path).await?;
+
+                let script_path = path!(&profile_path, "bin", NIXOS_SCRIPT);
+
+                run_script(script_path, Some(action.to_string()), NIXOS_SCRIPT).await?;
+            }
+        }
+
+        Ok(ExitCode::SUCCESS)
+    }
+}
+
+async fn run_script(
+    script_path: PathBuf,
+    action: Option<String>,
+    script_name: &str,
+) -> Result<(), FhError> {
+    tracing::debug!(
+        "Checking for {} script at {}",
+        script_name,
+        &script_path.display().to_string(),
+    );
+
+    if script_path.exists() && script_path.is_file() {
+        tracing::debug!(
+            "Found {} script at {}",
+            script_name,
+            &script_path.display().to_string(),
+        );
+
+        if let Ok(script_path_metadata) = tokio::fs::metadata(&script_path).await {
+            let permissions = script_path_metadata.permissions();
+            if permissions.mode() & 0o111 != 0 {
+                if let Some(action) = &action {
+                    tracing::info!("{} {}", &script_path.display().to_string(), action);
+                } else {
+                    tracing::info!("{}", &script_path.display().to_string());
+                }
+
+                let mut cmd = tokio::process::Command::new(&script_path);
+                if let Some(action) = action {
+                    cmd.arg(action);
+                }
+
+                let output = cmd
+                    .stdout(Stdio::inherit())
+                    .stderr(Stdio::inherit())
+                    .spawn()
+                    .wrap_err("failed to spawn Nix command")?
+                    .wait_with_output()
+                    .await
+                    .wrap_err(format!("failed to run {script_name} script"))?;
+
+                println!("{}", String::from_utf8_lossy(&output.stdout));
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn apply_path_to_profile(profile: &str, store_path: &str) -> Result<String, FhError> {
+    let profile_path = format!("/nix/var/nix/profiles/{profile}");
+
+    tracing::info!(
+        "Applying resolved store path {} to profile at {}",
+        store_path,
+        profile_path
+    );
+
+    nix_command(
+        &[
+            "build",
+            "--print-build-logs",
+            // `--max-jobs 0` ensures that `nix build` doesn't really *build* anything
+            // and acts more as a fetch operation
+            "--max-jobs",
+            "0",
+            "--profile",
+            &profile_path,
+            store_path,
+        ],
+        true,
```

Comment:
iirc this `true` means use sudo, but we shouldn't use sudo for `home-manager` -- let's hoist this up to a parameter based on the system type